editor/filesystem_dock.cpp

```diff
@@ -1721,6 +1736,18 @@ void FileSystemDock::_folder_removed(String p_folder) {
 		current_path = current_path.get_base_dir();
 	}
 
+	if (assigned_folder_colors.has(p_folder)) {
+		assigned_folder_colors.erase(p_folder);
+
+		if (!ProjectSettings::get_singleton()->has_setting("file_customization/folder_colors")) {
```

Comment:
This should probably be moved to a method. You repeat this code a couple of times.